frontend/app/utils/date-utils.ts

```diff
@@ -0,0 +1,60 @@
+import { useMemo } from 'react';
+
+import { padWithZero } from './string-utils';
+
+/**
+ * Parses a date string in the format "YYYY-MM-DD" and returns an object with the parsed components.
+ * @param date The date string to parse.
+ * @returns An object containing the parsed components (year, month, day). Returns an empty object if parsing fails or if the date does not exist.
+ */
+
+export function extractDateParts(date: string) {
+  const dateParts = date.split('-');
+
+  if (dateParts.length !== 3) {
+    return {};
+  }
+
+  const year = Number.parseInt(dateParts[0]);
+  const month = Number.parseInt(dateParts[1]);
+  const day = Number.parseInt(dateParts[2]);
+
+  if (isNaN(year) || isNaN(month) || isNaN(day)) {
+    return {};
+  }
+
+  if (!dateExists(year, month - 1, day)) {
+    return {};
+  }
+
+  return {
+    year: padWithZero(year, 4),
+    month: padWithZero(month, 2),
+    day: padWithZero(day, 2),
+  };
+}
+
+function dateExists(year: number, month: number, day: number): boolean {
+  const date = new Date(year, month, day);
+  return date.getFullYear() === year && date.getMonth() === month && date.getDate() === day;
+}
+
+/**
+ * Custom hook to retrieve an array of months based on the provided locale and format.
+ * @param locale - The locale to use for formatting the months.
+ * @param format - The format for displaying the months.
+ * @returns An array containing objects with month index and formatted month text.
+ */
+export function useMonths(locale: string, format: 'numeric' | '2-digit' | 'long' | 'short' | 'narrow' | undefined = 'long') {
+  return useMemo(() => {
```

Comment:
I don't think this should `useMemo()`. If the calling component needs to memoize it, then it should do so there, not in this function.

Reply:
I've done it in CDCP for perfromance. No need to call that factory logic orver and over. We should create a map or months/locale in `useMemo()` hook file instead to have 1 singleton per locale for optimization.

Reply:
In my opinion it's premature optimization. This isn't computationally intensive. And even if it was, the memoization should be done outside of the function.

Reply:
You look like a premature optimization. 🏃